src/github.com/getlantern/enproxy/lazyconn.go

```diff
@@ -46,6 +46,8 @@ func (l *lazyConn) get() (conn net.Conn, err error) {
 
 		// Wrap the connection in an idle timing one
 		l.connOut = idletiming.Conn(conn, l.p.IdleTimeout, func() {
+			l.p.connMapMutex.Lock()
+			defer l.p.connMapMutex.Unlock()
```

Comment:
@myleshorton This was a drive-by.  I noticed during testing that we weren't synchronizing access to this map on deleting a conn.